crates/common/logged_command/src/logged_command.rs

```diff
@@ -26,6 +54,86 @@ impl LoggingChild {
 
         outcome
     }
+    async fn update_and_log_outcome(
+        command_line: String,
+        outcome: Result<Output, std::io::Error>,
+        logger: &mut BufWriter<File>,
+        timeout: Duration,
+        status: &mut CmdStatus,
+    ) -> Result<Output, std::io::Error> {
+        let outcome = match status {
+            CmdStatus::Successful => outcome,
+            CmdStatus::KilledWithSigterm | CmdStatus::KilledWithSigKill => {
+                update_stderr_message(outcome, timeout)
+            }
+        };
+        if let Err(err) = LoggedCommand::log_outcome(&command_line, &outcome, logger).await {
+            error!("Fail to log the command execution: {}", err);
+        }
+        outcome
+    }
+
+    async fn timeout_operation(
+        status: &mut CmdStatus,
+        child_id: Option<u32>,
+        graceful_timeout: Duration,
+        forceful_timeout: Duration,
+    ) -> Result<(), std::io::Error> {
+        *status = CmdStatus::Successful;
+
+        tokio::time::sleep(graceful_timeout).await;
+
+        // stop the child process by sending sigterm
+        *status = CmdStatus::KilledWithSigterm;
+        send_sigal_to_stop_child(child_id, CmdStatus::KilledWithSigterm);
+        tokio::time::sleep(forceful_timeout).await;
+
+        // stop the child process by sending sigkill
+        *status = CmdStatus::KilledWithSigKill;
+        send_sigal_to_stop_child(child_id, CmdStatus::KilledWithSigKill);
+
+        // wait for the process to exit after signal
+        tokio::time::sleep(Duration::from_secs(120)).await;
+
+        Ok(())
+    }
+}
+
+fn update_stderr_message(
+    outcome: Result<Output, std::io::Error>,
+    timeout: Duration,
+) -> Result<Output, std::io::Error> {
+    let outcome = match outcome {
+        Ok(mut v) => {
+            v.stderr.append(
+                &mut format!(
+                    "operation failed due to timeout: duration={}s",
+                    timeout.as_secs()
+                )
+                .as_bytes()
+                .to_vec(),
+            );
+            Ok(v)
+        }
+        Err(e) => Err(std::io::Error::new(ErrorKind::Other, e.to_string())),
+    };
+
+    outcome
+}
```

Comment:
No don't hide the error behind a fake io error.

In fact it would be even simpler to take as parameter an input

```suggestion
fn update_stderr_message(
    mut output: Output
    timeout: Duration,
) -> Result<Output, std::io::Error> {
            output.stderr.append(
                &mut format!(
                    "operation failed due to timeout: duration={}s",
                    timeout.as_secs()
                )
                .as_bytes()
                .to_vec(),
            );

    output
}
```